src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs

```diff
@@ -0,0 +1,97 @@
+﻿using AutoMapper;
+using Microsoft.Dynamics.CRM;
+using Spd.Resource.Repository.Alias;
+using Spd.Resource.Repository.Application;
+using Spd.Resource.Repository.BizLicApplication;
+using Spd.Resource.Repository.LicApp;
+using Spd.Resource.Repository.Org;
+using Spd.Utilities.Dynamics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Spd.Resource.Repository.ControllingMemberCrcApplication;
+public class ControllingMemberCrcRepository : IControllingMemberCrcRepository
+{
+    private readonly DynamicsContext _context;
+    private readonly IMapper _mapper;
+
+
+    public ControllingMemberCrcRepository(IDynamicsContextFactory ctx, IMapper mapper)
+    {
+        _context = ctx.CreateChangeOverwrite();
+        _mapper = mapper;
+    }
+
+    public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMemberCrcApplicationAsync(CreateControllingMemberCrcAppCmd cmd, CancellationToken ct)
+    {
+        // get parrent business license application
+        spd_application? bizLicApplication = await _context.spd_applications
+                .Expand(a=>a.spd_businessapplication_spd_workerapplication)
+                .Expand(a => a.spd_ApplicantId_contact)
+                .Expand(a => a.spd_ApplicantId_account)
+                .Where(a => a.spd_applicationid == cmd.ParentBizLicApplicationId)
+                .SingleOrDefaultAsync(ct);
+
+        if (bizLicApplication == null)
+            throw new ArgumentException("Original business licence application was not found.");
+
+        // create controlling member application
+        spd_application app = _mapper.Map<spd_application>(cmd);
```

Comment:
where is mapping?

Reply:
you mean the file? in the ControllingMemberCrcApplication directory, under Repository.

Reply:
yes